algolia/opt/wait_configuration.go

```diff
@@ -0,0 +1,33 @@
+package opt
+
+import (
+	"time"
+)
+
+// WaitConfigurationOption stores the logic to change the delay between waits
+type WaitConfigurationOption struct {
+	DelayGrowth func(*time.Duration) time.Duration
+}
+
+// DefaultWaitConfiguration returns the new delay to apply between waits
+func DefaultWaitConfiguration() *WaitConfigurationOption {
+	return &WaitConfigurationOption{DelayGrowth: func(d *time.Duration) time.Duration {
+		const maxDurationBetweenWaits = 10 * time.Second
```

Comment:
could be added at the top level of the file to be consistent with the code base.